delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/query/ElasticSearchQueryError.scala

```diff
@@ -0,0 +1,54 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.query
+
+import akka.http.scaladsl.model.StatusCodes
+import ch.epfl.bluebrain.nexus.delta.kernel.utils.ClassUtils
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.JsonLdContext.keywords
+import ch.epfl.bluebrain.nexus.delta.sdk.error.ServiceError
+import ch.epfl.bluebrain.nexus.delta.sdk.http.HttpClientError
+import ch.epfl.bluebrain.nexus.delta.sdk.marshalling.HttpResponseFields
+import io.circe.syntax.KeyOps
+import io.circe.{Encoder, JsonObject}
+
+/**
+  * Enumeration of errors raised while querying the Elasticsearch indices
+  */
+sealed abstract class ElasticSearchQueryError(val reason: String) extends Product with Serializable
```

Comment:
The idea is to separate the rejections raised at query time from the other elasticsearch rejections